Доработана функциональность спринта по техническому заданию. Sprint 6 solution.

evgeniiriri opened the pull request.

evgeniiriri committed:
<feat> : Добавил свою реализацию связанного списка LinkedListTasks.java
в main есть небольшая проверка его.
коммит для переноса кода на домашнюю машину.

evgeniiriri committed:
перенос кода на домашний комп

evgeniiriri committed:
Написал тесты для LinkedListTasks, начал для нового функционала для InMemoryTaskManager.
перенос кода на домашний комп

evgeniiriri committed:
перенос кода

evgeniiriri committed:
<feature>: Истрия хранит только один вызов каждой задачи, в порядке их вызовов. Последний вызов в конце списка.
LinkedListTasks.java двусвязанный список, с базовыми возможностями и getTasks возвращает список задач в истории.
<test>: Тесты для новых функций.
В main написан сценарий по добавлению задач в историю, по удалению задач из истории.

evgeniiriri committed:
README.md добавил изменения.

Sla-als commented on src/kanban/service/InMemoryHistoryManager.java:
```diff
@@ -1,33 +1,51 @@
 public class InMemoryHistoryManager<T extends Task> implements HistoryManager<T> {
 
-    private final List<T> storage = new ArrayList<>();
+    private final LinkedListTasks<T> linkedListTasks = new LinkedListTasks<>();
```
В классе InMemoryHistoryManager, обратите внимание на использование обобщений (T extends Task). Это правильный подход для создания универсального менеджера истории, способного работать с различными типами задач. Однако убедитесь, что во всех местах, где используется T, корректно обрабатывается тип задачи. Например, в методе add, когда вы клонируете задачу, проверьте, что клонирование корректно работает для всех типов задач (Task, Epic, Subtask).

evgeniiriri replied:
Посмотрел в отладке и действительно, список подзадач у эпика и у клона эпика это один и тот же объект. Сделал у эпика дополнительное клонирование списка подзадач. В отладке показывает, что объекты разные.
Но вот как enum клонировать так и не разобрался, но в поставленных задачах статусы вычисляются или прописываются вместе с обновление или созданием задач.

Sla-als commented on src/kanban/test/InMemoryTaskManagerTest.java:
```diff
@@ -185,4 +164,32 @@ public void shouldIndependenceTasksAndHistory() {
         Assertions.assertNotEquals(task, inMemoryTaskManager.getHistory().get(0));
     }
 
+    @Test
```
Уделите внимание разработке тестов, которые покрывают различные сценарии использования вашего менеджера истории. важно проверить корректность работы при добавлении и удалении задач, а также при обработке специальных случаев, таких как добавление уже существующей в истории задачи

Sla-als reviewed (commented):
Здравствуйте! Ваша работа демонстрирует хорошее понимание задачи и умение работать с ключевыми концепциями Java, такими как коллекции, обобщения и управление памятью. Особенно хорошо продумана логика работы с историей задач, это важная часть задания. Оставил пару замечаний

evgeniiriri committed:
<fix>: скорректировал клонирование задачи для эпиков.
<test>: добавил тест удаления задач в менеджере и проверил, удаляются ли они в истории.
<test, docs>: добавил описание к тестам.

Sla-als commented:
Хорошая работа!
